src/swmmanywhere/graphfcns/design_graphfcns.py

```diff
@@ -15,102 +15,129 @@
+            hydraulic_design.depth_nbins,
+        ),
+    )
+
+
+def evaluate_design(
     ds_elevation: float,
     chamber_floor: float,
     edge_length: float,
     hydraulic_design: parameters.HydraulicDesign,
     Q: float,
-) -> tuple[float, float, float]:
-    """Design a pipe.
-
-    This function designs a pipe by iterating over a range of diameters and
-    depths. It returns the diameter and depth of the pipe that minimises the
-    cost function, while also maintaining or minimising feasibility parameters
-    associated with: surcharging, velocity and filling ratio.
+    diam: float,
+    depth: float,
+) -> dict[Hashable, float]:
+    """Evaluate the design of a pipe.
 
-    Cost equation from: https://doi.org/10.2166/hydro.2016.105
+    This function evaluates the design of a pipe by calculating the cost,
+    velocity, filling ratio, and feasibility parameters. It returns a dictionary
+    containing the design parameters and their values.
 
     Args:
-        ds_elevation (float): The downstream elevationq
+        ds_elevation (float): The downstream elevation
         chamber_floor (float): The elevation of the chamber floor
         edge_length (float): The length of the edge
         hydraulic_design (parameters.HydraulicDesign): A HydraulicDesign parameter
             object
         Q (float): The flow rate
-
-    Returns:
         diam (float): The diameter of the pipe
         depth (float): The depth of the pipe
-        cost (float): The cost of the pipe
+
+    Returns:
+        dict: A dictionary containing the designed pipe's parameters and values
     """
-    designs = product(
-        hydraulic_design.diameters,
-        np.linspace(
-            hydraulic_design.min_depth, hydraulic_design.max_depth, 10
-        ),  # TODO should 10 be a param?
+    A = np.pi * diam**2 / 4
+    n = 0.012  # mannings n
+    R = A / (np.pi * diam)  # hydraulic radius
+    # TODO... presumably need to check depth > (diam + min_depth)
+
+    elev_diff = chamber_floor - (ds_elevation - depth)
+    slope = elev_diff / edge_length
+    # Always pick a pipe that is feasible without surcharging
+    # if available
+    surcharge_feasibility = 0.0
+    # Use surcharged elevation
+    while slope <= 0:
+        surcharge_feasibility += 0.05
+        slope = (
+            chamber_floor + surcharge_feasibility - (ds_elevation - depth)
+        ) / edge_length
+        # TODO could make the feasibility penalisation increase
+        # when you get above surface_elevation[node]... but
+        # then you'd need a feasibility tracker and an offset
+        # tracker
+    v = (slope**0.5) * (R ** (2 / 3)) / n
+    filling_ratio = Q / (v * A)
+    # buffers from: https://www.polypipe.com/sites/default/files/Specification_Clauses_Underground_Drainage.pdf
+    average_depth = (depth + chamber_floor) / 2
+    V = edge_length * (diam + 0.3) * (average_depth + 0.1)
+    cost = 1.32 / 2000 * (9579.31 * diam**0.5737 + 1163.77 * V**1.31)
```

Comment:
I'm sure these numbers are perfectly fine, but as they are now, they are just somewhat random numbers coming from nowhere.

This function could probably be split down into small chunks, but even if you don't want to go down that route, yet, I would at least separate the cost calculation into its own function, explaining in the docstring what this numbers are, where they are coming from, etc.